counsel.el

```diff
@@ -4656,11 +4656,12 @@ selected color."
 
 ;;** `counsel-linux-app'
 (defcustom counsel-linux-apps-directories
-  '("~/.local/share/applications/"
-    "~/.guix-profile/share/applications/"
-    "/usr/local/share/applications/"
-    "/var/lib/flatpak/exports/share/applications/"
-    "/usr/share/applications/")
+  (let ((xdg-data-dirs (getenv "XDG_DATA_DIRS")))
+    (mapcar (lambda (dir) (concat dir "/applications"))
+            (split-string (if (and xdg-data-dirs (not (string= xdg-data-dirs "")))
```

Comment:
Obviously either way is fine, but just in case you prefer: the negation of this condition is `(member xdg-data-dirs '(nil ""))`.